suite/suite_test.go

```diff
@@ -617,3 +617,47 @@ func (s *FailfastSuite) Test_B_Passes() {
 	s.call("Test B Passes")
 	s.Require().True(true)
 }
+
+type ShuffleOrderSuite struct {
+	Suite
+	callOrder []string
+}
+
+func (s *ShuffleOrderSuite) call(method string) {
+	s.callOrder = append(s.callOrder, method)
+}
+
+func TestSuiteShuffleOrder(t *testing.T) {
+	Run(t, new(ShuffleOrderSuite))
+}
+
+func TestShuffleOrderSuiteShuffleOn(t *testing.T) {
+	// To test this with testify.shuffle on we launch it in its own process
+	cmd := exec.Command("go", "test", "-v", "-race", "-run", "TestSuiteShuffleOrder", "-testify.shuffle", "2")
```

Comment:
Add a comment mentionning that the test reliability relies on the algorithm used by package `math/rand`. If that algorithm changes the seed value may have to be adapted.